### PR TITLE
Implement np.argwhere & np.flatnonzero

### PR DESCRIPTION
**Pull request by jakevdp**
#2080

### CODE REVIEW

File: jax/numpy/lax_numpy.py [hunk 1]
```diff
@@ -2941,6 +2946,14 @@ def vander(x, N=None, increasing=False):
 ### Misc
 
 
+@_wraps(np.argwhere)
+def argwhere(a):
+  result = transpose(vstack(nonzero(a)))
```
Comment (hawkinsp) [inline jax/numpy/lax_numpy.py:2951]: I'm curious if the `vstack` is needed, given the NumPy docs say "np.argwhere(a) is almost the same as np.transpose(np.nonzero(a)), but produces a result of the correct shape for a 0D array."

**Review by hawkinsp — APPROVED** [review]

File: jax/numpy/lax_numpy.py (hunk 1 above)
Comment (jakevdp) [reply jax/numpy/lax_numpy.py:2951]: nonzero returns a tuple of 1D arrays, which ``np.transpose`` silently converts to a 2D array. Jax returns an error without the explicit vstack.

**Comment by jakevdp** [comment]
Looks like the test is not stable across numpy versions:
```python
>>> import numpy as np
>>> print(np.__version__)
>>> print(np.argwhere(0).shape)
1.16.4
(0, 1)
```
```
>>> import numpy as np
>>> print(np.__version__)
>>> print(np.argwhere(0).shape)
1.18.4
(0, 0)
```
I'll skip relevant tests and try again.

**Comment by jakevdp** [comment]
Internal tests passing

**Pull request merged**